mailpoet/assets/js/src/automation/listing/index.tsx

```diff
@@ -1,17 +1,90 @@
 import { Search, TableCard } from '@woocommerce/components/build';
+import { TabPanel } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { useHistory, useLocation } from 'react-router-dom';
 import { getRow } from './get-row';
-import { Workflow } from './workflow';
+import { Workflow, WorkflowStatus } from './workflow';
 
 type Props = {
   workflows: Workflow[];
   loading: boolean;
 };
 
+type FilterTab = {
+  name: string;
+  title?: JSX.Element;
+  classname: string;
+};
+
 export function AutomationListing({ workflows, loading }: Props): JSX.Element {
-  const rows = workflows.map((workflow) => getRow(workflow));
+  const history = useHistory();
+  const location = useLocation();
+  const pageSearch = new URLSearchParams(location.search);
+
+  const updateUrlSearchString = (keysAndValues: Record<string, string>) => {
+    Object.keys(keysAndValues).forEach((key) => {
+      pageSearch.set(key, keysAndValues[key]);
+      if (['per_page', 'status'].includes(key)) {
+        pageSearch.set('paged', '1');
+      }
+    });
+    history.replace({ search: pageSearch.toString() });
+  };
```

Comment:
Just to be sure, I'd wrap this in `useCallback()` so the function doesn't get redefined with each render. Not really important though, I guess.

Reply:
I made this change, and also added `useMemo` for the `pageSearch` variable, but I'm not sure how much of a difference it will make. Won't it get redefined every time anyway since it will have `pageSearch` in its dependency array? Maybe it's useful to have for later, when we add other features and wouldn't want to redefine this callback if they make changes that don't affect the query string for the page?

Reply:
Thank you very much for this detailed explanation!